### PR TITLE
Execute read operations in computation thread

### PR DESCRIPTION
**Pull request by apoi**

### CODE REVIEW

File: reark/src/main/java/io/reark/reark/data/store/ContentProviderStore.java [hunk 1]
```diff
@@ -107,17 +108,21 @@ protected void insertOrUpdate(T item, Uri uri) {
         return list;
     }
 
-    @Nullable
-    protected T queryOne(Uri uri) {
-        final List<T> queryResults = queryList(uri);
+    @NonNull
+    protected Observable<T> queryOne(Uri uri) {
+        return Observable.just(uri)
```
Comment (jaggernod) [inline reark/src/main/java/io/reark/reark/data/store/ContentProviderStore.java]: I might have a bit better way of doing it. Can write it next weekend

Comment (apoi) [inline reark/src/main/java/io/reark/reark/data/store/ContentProviderStore.java]: This was an adaptation of the existing code and felt kinda hackish, so sounds good :+1:

**Comment by tehmou** [comment]
@apoi perhaps we merge this and @jaggernod can put his own later?

**Comment by apoi** [comment]
Yes, I would suggest we do that.

**Pull request merged**